lang/rust/avro/src/decimal.rs

```diff
@@ -214,4 +218,63 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn other() -> TestResult {
+        let schema_str = r#"
+    {
+      "type": "record",
+      "name": "test",
+      "fields": [
+        {
+          "name": "field_name",
+          "type": "bytes",
+          "logicalType": "big-decimal"
+        }
+      ]
+    }
+    "#;
+        let schema = Schema::parse_str(schema_str)?;
+
+        // build record with big decimal value
+        let mut record = Record::new(&schema).unwrap();
+        let val = BigDecimal::new(BigInt::from(12), 2);
+        record.put("field_name", val.clone());
+
+        // save record in writer
+        let codec = Codec::Null;
+        let mut writer = Writer::builder()
+            .schema(&schema)
+            .codec(codec)
+            .writer(Vec::new())
+            .build();
+
+        let result = writer.append(record.clone());
+        assert!(result.is_ok(), "Append KO  : {}", result.unwrap_err());
+        assert!(writer.flush().is_ok(), "Flush KO");
```

Comment:
Actually there is no need to `assert!()` these above. You could just use `?`, e.g. `writer.append(record.clone())?;` and it will propagate the error because the method returns `TestResult`.

I try to avoid using `.unwrap()` in general. Especially in non-test code its usage should be forbidden. I am not sure whether there is a Clippy check that we could enforce.

Reply:
https://doc.rust-lang.org/book/ch09-03-to-panic-or-not-to-panic.html#cases-in-which-you-have-more-information-than-the-compiler